### ISSUE TITLE
PID Autotune cancels M104/M109 on other extruder/bed

### ISSUE THREAD

**Issue by GreatGrizzly** [MarlinFirmware/Marlin#4848]
When I issue a M190 to heat up my bed, then run PID Autotune on the Hot End, the bed temperature changes back to zero during the process. While this is hardly disastrous, the extra time it takes to reheat the bed can be a problem for a leaky hot end that is waiting on the print bed.

Here is my slic3r start code:

``` gcode
;If printer is waiting for heatup (M109) or bed heatup (M190), break out of the wait.
M108

;Save settings just in case I forget.
M500

;Wait for bed temperature to get to target temperature.
M190 R[first_layer_bed_temperature]

;Home all Axis
G28

;Move print head to back left to get it out of the way.
G0 X0 Y245 Z5 F2000

;**PID AutoTune will cancel the M190 for the bed, even though I am AutoTuning the hot end.**
M303 E0 S[temperature] C4 U1

;Save settings just in case
M500

;The below code is used to correct for the PID Autotune bug above.
;hot end reaches target temperature, continue on.
M104 R[first_layer_temperature]

;Wait for bed temperature to get to target temperature, again...
M190 R[first_layer_bed_temperature]
```

I am using RCBugFix code updated yesterday.

**Comment by FHeilmann** [comment]
Why do you run a PID autotune before every print?

Autotune is meant to be run once when you change Voltage/Heating hardware. The results are committed to EEPROM and are not in need of changing unless you change hardware again or something about your heating process is getting weird due to wear or other external influences. It is not necessary to re-run the autotune for each print.

**Comment by GreatGrizzly** [comment]
From my understanding: PID values change depending on the target
temperature of the hot end/bed. If I have different types a filaments at
different temperatures, wouldn't it make sense to PID tune the hot end/bed
for each one? I want to introduce as much automation as possible.

My thinking is that most of the wait time before a print is for the warm up
of the hot end/bed, so doing a PID Autotune cycle is very small increase in
time for improved convenience and PID accuracy.

**Comment by Roxy-3D** [comment]
Well...  What you say is true.    But the PID algorithms are not ridiculously sensitive.   They are pretty stable.   If you load up an extruder with a Black PLA filament and run the autotune at 205 C. , you will get a set of numbers.   If you switch the nozzle to a White ABS (and purge it) and autoturn at 230 C. you will get another set of numbers.

You can usually take the middle ground and load that into your Configuration.h and be very happy with the results regardless of what you are printing with.

Remember...   The autotune numbers are not 100% accurate.  They are measuring and fine tuning the nozzle heating for no extrusion of filament.   That is different than real life with filament being pushed through through the nozzle.

**Comment by GreatGrizzly** [comment]
Another thing about turning on AutoTune on each print is over time (assuming you 
have the same filament and temperature), the PID settings will get more 
accurate on each run (up to a max), as it is using the old results to get the new 
results which will probably be very similar. I have found that 
filaments like PETG are extremely sensitive to 1 degree variations, so 
taking the middle ground might not be a viable option. Especially when I 
switch between PLA and PETG, 195c vs 250c.

This defect is hardly a show stopper, so is low priority.

**Comment by Roxy-3D** [comment]
> I have found that filaments like PETG are extremely sensitive to 1 degree variations, so
> taking the middle ground might not be a viable option. 

For the 'very sensitive' filaments, it might be reasonable to actually extrude some filament  during the tuning process.   It could be argued how much of that needs to be done, but for sure that is going to affect what the 'optimum parameters' are.

**Comment by GreatGrizzly** [comment]
That is a very good idea. Usually when I autotune, I have the filament in
the chamber.

**Comment by brainscan** [comment]
I've only needed to use auto tune once and my nozzle temp is always solid regardless of which filament. What hotend do you use maybe if you're having problems it's hardware related as I've never known a filament to have a 1° operating window.

**Comment by thinkyhead** [comment]
You'll note also that when you run filament faster through the extruder, it cools down more quickly. Same thing if your spool of filament is a few degrees cooler on a given day. The `M303` autotune feature is meant to provide stability for the bulk of the thermal mass in the hot-end, and not to deal with these kinds of variations.

You should try using the `AUTOTEMP` feature. This causes the heater to adjust itself adjust based on the volume of material being pushed into the melt chamber. It allows you to set a factor, which you can base on the kind of material you're using and the ambient temperature.
